src/index.js

```diff
@@ -12,9 +12,11 @@ makeServer();
 ReactDOM.render(
   <React.StrictMode>
     <BrowserRouter>
-      <ProductProvider>
-        <App />
-      </ProductProvider>
+      <AuthProvider>
```

Comment:
Product Provider should be outside as it is a super-set.It will not change anything but product context is not dependent on auth Context.
In Future Auth Context should be on top of Cart and Wishlist as both are dependent on Auth Context

Reply:
Will implement this one ahead